etcdctl/snapshot/v3_snapshot.go

```diff
@@ -275,32 +271,54 @@ func (s *v3Manager) Restore(cfg RestoreConfig) error {
 	return nil
 }
 
+func (s *v3Manager) outDbPath() string {
+	return filepath.Join(s.snapDir, "db")
+}
+
 // saveDB copies the database snapshot to the snapshot directory
 func (s *v3Manager) saveDB() error {
-	f, ferr := os.OpenFile(s.dbPath, os.O_RDONLY, 0600)
+	err := s.copyAndVerifyDB()
+	if err != nil {
+		return err
+	}
+
+	be := backend.NewDefaultBackend(s.outDbPath())
+	defer be.Close()
+
+	err = membership.TrimMembershipFromBackend(s.lg, be)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *v3Manager) copyAndVerifyDB() error {
+	srcf, ferr := os.OpenFile(s.srcDbPath, os.O_RDONLY, 0600)
```

Comment:
Looks like it is missing `os.O_CREATE` flag ([flags docs](https://golang.org/pkg/os/#pkg-constants)). I'd suggest 
```
srcf, ferr := os.OpenFile(s.srcDbPath, os.O_RDONLY||os.O_CREATE, 0600)
```
that is [documented here](https://golang.org/pkg/os/#OpenFile) if we want to create file (in case of non-existance) or
```
srcf, ferr := os. Open(s.srcDbPath)
```
[documented here](https://golang.org/pkg/os/#Open) if we only need to open file

Reply:
Done.